src/controllers/documents.controller.js

```diff
@@ -14,9 +16,27 @@ exports.getAll = async (req, res, next) => {
       documents: userInfo.pdfDocuments,
     });
   } catch (error) {
-    error.message = "Internal Server Error";
+    error.message = ERRORMESSAGE.ERROR_500;
     error.status = 500;
 
     next(error);
   }
 };
+
+exports.getDocument = async (req, res, next) => {
+  try {
+    const { userId, documentId } = req.params;
+    const pdfDocument = await getDocumentInS3(userId, documentId, next);
+    res.set({
+      "Content-Type": "application/pdf",
+      "Content-Length": pdfDocument.length,
```

Comment:
```js
"Content-Length": pdfDocument.length,
```
pdfDocument의 길이를 헤더에 보내는 이유가 궁금합니다.

Reply:
> There are several advantages of including the Content-Length header:
> 
> Improved performance: By specifying the size of the response body, the client can allocate the necessary resources to receive the entire response in one go, rather than having to wait for multiple smaller chunks of the response to arrive. This can improve performance and reduce the number of network requests required.
> 
> Better error handling: If the server sends a response with no Content-Length header, the client may not know when the response is complete and may continue to wait for more data, resulting in a timeout or other error. Including the Content-Length header ensures that the client knows when it has received the complete response, allowing for better error handling.
> 
> Easier resource allocation: The Content-Length header allows the client to allocate the necessary memory or buffer space for the response, which can be especially important for large responses such as PDF files. By knowing the size of the response in advance, the client can ensure that it has enough memory to receive and process the entire response.
> 
> Overall, including the Content-Length header when sending a PDF file or any other type of response from an Express server can help to improve performance, reduce errors, and make it easier to allocate necessary resources

간단하게 요약하면,
1. 헤더에 Content-Length를 명시함으로서, 클라이언트에서 여러개 의 청크가 도착할 때 까지 기다릴 필요 없이, 한번에 전체 응답을 수신하는 데 필요한 리소스를 확인 후 메모리를 할당 할 수 있어 성능을 향상시킬 수 있다고합니다.

2. Content-Length를 명시함으로서, 클라이언트는 완전한 응답 수신 시기를 확인 할 수 있고, 더 나은 오류 처리를 할 수 있다고 합니다.
=> 타임아웃 에러를 기다리는 대신 길이 정보를 확인해, 응답의 완료 여부를 확인할 수 있기 때문.